crates/ra_ide_api/src/expand_macro.rs

```diff
@@ -49,7 +49,14 @@ fn expand_macro_recur(
         let node = hir::Source::new(macro_file_id, &child);
         let new_node = expand_macro_recur(db, source, node)?;
 
-        replaces.insert(child.syntax().clone().into(), new_node.into());
+        // Replace the whole node if it is root
+        // `replace_descendants` will not replace the parent node
+        // but `SyntaxNode::descendants include itself
+        if expanded == *child.syntax() {
```

Comment:
Or should we fix `replace_descendants` instead ?

Reply:
No, I think this is a better fix